### ISSUE TITLE
Hyperparameter values vs Rank

### ISSUE THREAD

**Issue by shaddyab** [hyperopt/hyperopt#647]
If after running hyperopt for a large number of iterations (e.g., 1000 iterations) and plotting the values of each hyperparameter in the search space vs the rank (i.e., loss ) I notice that the values of some of the hyperparameters indeed converge to a constant value as a function of rank, while other hyperparameter never converge to a constant value, can I conclude that the parameters which never converge are not important to the model and it is better be fixed their value rather that including them in the search space?

**Comment by bjkomer** [comment]
That may be the case, but isn't the only possible conclusion based on your results. A hyperparameter that never converges could either be not important, or just highly dependent on the settings of other hyperparameters.

This is demonstrated by the toy example of `loss = (a + b) ** 2`. Looking at hyperparemeters `a` or `b` individually, you likely won't see clear convergence to a constant (any value could give you the lowest possible loss), since the actual best setting is the relationship a = -b. You'll have to plot the parameters with respect to others to see this.

That being said, I don't know how common this kind of relationship is in real problems, it may be more likely that they are just unimportant.

**Comment by shaddyab** [comment]
@bjkomer Thank you for taking the time to answer my question. Your explanation makes sense to me.
To be more specific, I am tuning an XGBoost model where I noticed that parameters such as max_depth, n_estimators,  and learning_rate will converge. While regularization parameters such as reg_alpha, and reg_lambda will not converge. I think it is because they are highly dependent and one of them need to be fixed (maybe reg_alpha) and the other tuned.
